plugins/woocommerce/src/Blocks/Shipping/ShippingController.php

```diff
@@ -433,6 +430,42 @@ function ( $rate ) {
 		return $packages;
 	}
 
+	/**
+	 * Remove shipping if no address is entered.
+	 *
+	 * @param array $packages Array of shipping packages.
+	 * @return array
+	 */
+	public function remove_shipping_if_no_address( $packages ) {
+		$customer = wc()->customer;
+		$has_address = $customer->has_shipping_address();
+
+		remove_filter( 'option_woocommerce_shipping_cost_requires_address', array( $this, 'override_cost_requires_address_option' ) );		
+		$option_checked = filter_var( get_option( 'woocommerce_shipping_cost_requires_address', 'no' ), FILTER_VALIDATE_BOOLEAN );
+		add_filter( 'option_woocommerce_shipping_cost_requires_address', array( $this, 'override_cost_requires_address_option' ) );
+
+		if ( ! $has_address && $option_checked ) {
+			$packages = array_map(
+				function ( $package ) {
+					if ( ! is_array( $package['rates'] ) ) {
```

Comment:
Do we want to check if `$package['rates']` is set first before we try is_array on it, just thinking if $package doesn't have a member called rates this will warn, won't it?